pimcore/lib/Pimcore/Bundle/CoreBundle/Command/InternalNewsletterDocumentSendCommand.php

```diff
@@ -63,12 +64,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $addressSourceAdapterName = $data['addressSourceAdapterName'];
         $adapterParams = $data['adapterParams'];
 
-        $adapterClass = '\\Pimcore\\Document\\Newsletter\\AddressSourceAdapter\\' . ucfirst($addressSourceAdapterName);
+        $serviceLocator = $this->getContainer()->get('pimcore.newsletter.address_source_adapter.factories');
+
+        if (!$serviceLocator->has($addressSourceAdapterName)) {
+            //TODO: Shouldn't this throw an exception?
```

Comment:
I would definitely throw an exception here. Previously a class not found exception was thrown, now we just can use a custom one.